src/Components/AppDownloadFooter.tsx

```diff
@@ -37,20 +39,36 @@ export const AppDownloadFooter: FC<AppDownloadFooterProps> = () => {
   const ref = useRef<HTMLDivElement | null>(null)
 
   const { downloadAppUrl } = useDeviceDetection()
+  const { match } = useRouter()
 
-  useRouteComplete({
-    // Used to ensure we just show the banner after an initial internal navigation
-    onComplete: async () => {
-      if (!ref.current || !("animate" in ref.current)) return
+  const isVisible = useRef(false)
+  const routeChanges = useRef(-1)
+
+  useEffect(() => {
+    const showBanner = async () => {
+      if (!ref.current || !("animate" in ref.current) || isVisible.current) {
+        return
+      }
 
       const animation = ref.current.animate(
         [{ transform: "translateY(100%)" }, { transform: "translateY(0)" }],
         { duration: 250, easing: "ease-out", fill: "forwards" }
       )
 
       await animation.finished
-    },
-  })
+      isVisible.current = true
+    }
+
+    // Increment route changes counter when pathname changes
+    if (match.location.pathname) {
+      routeChanges.current++
+    }
+
+    // Show banner after second route change
+    if (routeChanges.current === 2) {
+      showBanner()
+    }
+  }, [match.location.pathname])
```

Comment:
This approach makes sense but would it work to detect elements (meaning, if the page is loading)? This is the way we can always see if things are fetching / showing top bar or not:

```
const { match } = useRouter() 

if (!match.elements) {
  // is fetching 
}

if (match.elements) {
  // page is done loading
}
```

Reply:
That's what was happening here previously that doesn't work for cache pages. It will never fire, because it will never be loading.

Reply:
makes sense. I wonder how we can generalize this a bit more, but seems a bit fussy...